Gems/EMotionFX/Code/EMotionFX/Source/MotionSet.cpp

```diff
@@ -726,10 +726,15 @@ namespace EMotionFX
     {
         MCore::LockGuardRecursive lock(m_mutex);
 
-        return AZStd::accumulate(begin(m_childSets), end(m_childSets), size_t{0}, [](size_t total, const MotionSet* motionSet)
+        size_t childSetSize = 0;
+        for (const MotionSet* motionSet : m_childSets)
         {
-            return total + motionSet->GetIsOwnedByRuntime();
-        });
+            if (!motionSet->GetIsOwnedByRuntime())
+            {
+                ++childSetSize;
+            }
+        }
+        return childSetSize;
```

Comment:
As far as I can tell, these two implementations of this method should produce identical output.
https://godbolt.org/z/3467n15Pr

Can you explain how the old version is incorrect? Are there other uses of `std::accumulate` that we need to update?

Reply:
is it always safe to add a size_t to a boolean value?

Reply:
Checked into it: The issue wasn't the accumulate, but a missing "!" which increased the value in wrong cases. I've still converted back to the old loop but used the add rather than the if, as this results in less code and is simple.